src/matter/session/secure/SecureChannelMessenger.ts

```diff
@@ -25,9 +39,18 @@ export class SecureChannelMessenger<ContextT> {
         return schema.decode((await this.nextMessage(expectedMessageType)).payload);
     }
 
+    async waitForStatusResponse() {
+        const response = await this.nextMessage(MessageType.StatusReport);
+        return this.decodeStatusReport(response.payload);
+    }
+
     async waitForSuccess() {
-        // If the status is not Success, this would throw an Error.
-        await this.nextMessage(MessageType.StatusReport);
+        const { generalStatus, protocolStatus } = await this.waitForStatusResponse();
```

Comment:
If throwIfError is throwing a ChannelStatusResponseError, this is equivalent to the original code.
Not sure why you need this change here...